Edit primary branch in IDE release note and docs changes (#4972)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/release-notes/72-Feb-2024/ide-git-edit.md

---
title: "New: Edit in primary git branch in IDE"
description: "February 2024: Edit access to the primary git branch while in the IDE. This was previously a read-only feature."
sidebar_label: "New: Edit primary git branch in IDE"
sidebar_position: 06
tags: [Feb-2024]
date: 2024-02-22
---

> You can now edit the primary branch of your git repo in the dbt Cloud IDE. The primary branch of the connected git repo has traditionally been _read-only_ in the IDE. This update changes the branch to _protected_ and allows direct edits. When a commit is made, dbt Cloud will prompt you to create a new branch. dbt Cloud will pre-populate the new branch name with the USERNAME-patch-#; however, you can edit the field with a custom branch name.

Author: ```suggestion
You can now edit, format, or lint files and execute dbt commands directly in your primary git branch in the dbt Cloud IDE.  This enhancement is available across various repositories: native integrations, imported git URLs, and managed repos.

The primary branch of the connected git repo has traditionally been _read-only_ in the IDE. This update changes the branch to _protected_ and allows direct edits. When a commit is made, dbt Cloud will prompt you to create a new branch. dbt Cloud will pre-populate the new branch name with the USERNAME-patch-#; however, you can edit the field with a custom branch name.
```

---

> title: "New: Edit in primary git branch in IDE"

Author: ```suggestion
title: "Enhancement: Edit in primary git branch in IDE"
```

description: "February 2024: Edit access to the primary git branch while in the IDE. This was previously a read-only feature."

> sidebar_label: "New: Edit primary git branch in IDE"

Author: ```suggestion
sidebar_label: "Enhancement: Edit primary git branch in IDE"
```

---
title: "Enhancement: Edit in primary git branch in IDE"

> description: "February 2024: Edit access to the primary git branch while in the IDE. This was previously a read-only feature."

Reviewer: i think we can clarify that they can commit to a new branch, just so it's super clear they can't commit changes in their main branch.

```suggestion
description: "February 2024: Edit directly on your primary git branch and commit those changes to a new branch when ready. This was previously a read-only feature." 
```

Reviewer: wdyt?

Reviewer: i want to make sure the description is super clear when it's announced in the slack community
<img width="920" alt="Screenshot 2024-02-23 at 15 17 31" src="https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/4aaf2c51-6f4b-4a8a-b39a-4e8e949429ea">

> description: "February 2024: Edit directly on your primary git branch and commit those changes to a new branch when ready. This was previously a read-only feature." 

Author: ```suggestion
description: "February 2024: Edit directly on your primary git branch and create a new branch when ready to commit. The primary branch was previously read-only but now allows editing while remaining protected." 
```

Reviewer: this reads lovely!

sidebar_label: "Enhancement: Edit primary git branch in IDE"
sidebar_position: 06
tags: [Feb-2024]

> date: 2024-02-22

Reviewer: ```suggestion
date: 2024-02-23
```

Reviewer: today's date

---

> You can now edit, format, or lint files and execute dbt commands directly in your primary git branch in the dbt Cloud IDE.  This enhancement is available across various repositories: native integrations, imported git URLs, and managed repos.

Reviewer: ```suggestion
You can now edit, format, or lint files and execute dbt commands directly in your primary git branch in the [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud).  This enhancement is available across various repositories: native integrations, imported git URLs, and managed repos.
```

> You can now edit, format, or lint files and execute dbt commands directly in your primary git branch in the [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud).  This enhancement is available across various repositories: native integrations, imported git URLs, and managed repos.

Author: ```suggestion
You can now edit, format, or lint files and execute dbt commands directly in your primary git branch in the [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud).  This enhancement is available across various repositories, including native integrations, imported git URLs, and managed repos.
```

> The primary branch of the connected git repo has traditionally been _read-only_ in the IDE. This update changes the branch to _protected_ and allows direct edits. When a commit is made, dbt Cloud will prompt you to create a new branch. dbt Cloud will pre-populate the new branch name with the USERNAME-patch-#; however, you can edit the field with a custom branch name.

Reviewer: ```suggestion
The primary branch of the connected git repo has traditionally been _read-only_ in the IDE. This update changes the branch to _protected_ and allows direct edits. When a commit is made, dbt Cloud will prompt you to create a new branch. dbt Cloud will pre-populate the new branch name with the GIT_USERNAME-patch-#; however, you can edit the field with a custom branch name.
```

Previously, the primary branch was displayed as read-only, but now the branch is displayed with a lock icon to indentify it as protected:

<DocCarousel slidesPerView={1}>

> <Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/read-only.png" width="85%" title="Previous read-only experience"/>

Reviewer: ```suggestion
<Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/read-only.png" width="75%" title="Previous read-only experience"/>
```

> <Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/protected.png" width="85%" title="New protected experience"/>

Reviewer: ```suggestion
<Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/protected.png" width="75%" title="New protected experience"/>
```

</DocCarousel>

> When you make a commit while on the primary branch, a model window will open prompting you to create a new branch and enter a commit message:

Reviewer: ```suggestion
When you make a commit while on the primary branch, a modal window will open prompting you to create a new branch and enter a commit message:
```

> <Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/create-new-branch.png" width="90%" title="Create new branch window"/>

Reviewer: ```suggestion
<Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/create-new-branch.png" width="75%" title="Create new branch window"/>
```
